server/api/auth_handler.py

```diff
@@ -0,0 +1,152 @@
+        if not user:
+            try:
+                if len(post_data.get("password")) < 6:
+                    responseObject = {
+                        "status": "fail",
+                        "message": "Password must be at least six characters long."
+                    }
+                    return make_response(jsonify(responseObject)), 401
+                if post_data.get("password") != post_data.get("confirm"):
+                    responseObject = {
+                        "status": "fail",
+                        "message": "Passwords must match."
+                    }
+                user = User(
+                    email=post_data.get("email"),
+                    password=post_data.get("password")
+                )
+                db.session.add(user)
+                db.session.commit()
+                auth_token = user.encode_auth_token(user.id)
+                g.user = user
+                responseObject = {
+                    "status": "success",
+                    "message": "Successfully registered.",
+                    "auth_token": auth_token.decode()
+                }
+                return make_response(jsonify(responseObject)), 201
+            except Exception as e:
+                responseObject = {
+                    "status": "fail",
+                    "message": "Some error occurred. Please try again."
+                }
+                return make_response(jsonify(responseObject)), 401
+        else:
+            responseObject = {
+                "status": "fail",
+                "message": "User already exists. Please log in."
+            }
+            return make_response(jsonify(responseObject)), 202
+
+
+class LoginAPI(MethodView):
+    def post(self):
+        post_data = request.get_json()
+        try:
+            user = User.query.filter_by(
+                email=post_data.get("email")
+            ).first()
+            if user and flask_bcrypt.check_password_hash(
+                user.password, post_data.get("password")
+            ):
+                auth_token = user.encode_auth_token(user.id)
+                if auth_token:
+                    g.user = user
+                    responseObject = {
+                        "status": "success",
+                        "message": "Successfully logged in",
+                        "auth_token": auth_token.decode()
+                    }
+                    return make_response(jsonify(responseObject)), 200
+                else:
+                    responseObject = {
+                        "status": "fail",
+                        "message": "User does not exist."
+                    }
+                    return make_response(jsonify(responseObject)), 404
+        except Exception as e:
+            print(e)
+            responseObject = {
+                "status": "fail",
+                "message": "Try again"
+            }
+            return make_response(jsonify(responseObject)), 500
+
```

Comment:
Can we have more specific error messages based on 1)if user does not exist and 2) password does not match the user?